entities/vectorindex/hnsw/config.go

```diff
@@ -109,82 +109,91 @@ func ParseAndValidateConfig(input interface{}) (schema.VectorIndexConfig, error)
 
 	asMap, ok := input.(map[string]interface{})
 	if !ok || asMap == nil {
-		return uc, fmt.Errorf("input must be a non-nil map")
+		return nil, fmt.Errorf("input must be a non-nil map")
 	}
 
 	if err := optionalIntFromMap(asMap, "maxConnections", func(v int) {
 		uc.MaxConnections = v
 	}); err != nil {
-		return uc, err
+		return nil, err
 	}
 
 	if err := optionalIntFromMap(asMap, "cleanupIntervalSeconds", func(v int) {
 		uc.CleanupIntervalSeconds = v
 	}); err != nil {
-		return uc, err
+		return nil, err
 	}
 
 	if err := optionalIntFromMap(asMap, "efConstruction", func(v int) {
 		uc.EFConstruction = v
 	}); err != nil {
-		return uc, err
+		return nil, err
 	}
 
 	if err := optionalIntFromMap(asMap, "ef", func(v int) {
 		uc.EF = v
 	}); err != nil {
-		return uc, err
+		return nil, err
 	}
 
 	if err := optionalIntFromMap(asMap, "dynamicEfFactor", func(v int) {
 		uc.DynamicEFFactor = v
 	}); err != nil {
-		return uc, err
+		return nil, err
 	}
 
 	if err := optionalIntFromMap(asMap, "dynamicEfMax", func(v int) {
 		uc.DynamicEFMax = v
 	}); err != nil {
-		return uc, err
+		return nil, err
 	}
 
 	if err := optionalIntFromMap(asMap, "dynamicEfMin", func(v int) {
 		uc.DynamicEFMin = v
 	}); err != nil {
-		return uc, err
+		return nil, err
 	}
 
 	if err := optionalIntFromMap(asMap, "vectorCacheMaxObjects", func(v int) {
 		uc.VectorCacheMaxObjects = v
 	}); err != nil {
-		return uc, err
+		return nil, err
 	}
 
 	if err := optionalIntFromMap(asMap, "flatSearchCutoff", func(v int) {
 		uc.FlatSearchCutoff = v
 	}); err != nil {
-		return uc, err
+		return nil, err
 	}
 
 	if err := optionalBoolFromMap(asMap, "skip", func(v bool) {
 		uc.Skip = v
 	}); err != nil {
-		return uc, err
+		return nil, err
 	}
 
 	if err := optionalStringFromMap(asMap, "distance", func(v string) {
 		uc.Distance = v
 	}); err != nil {
-		return uc, err
+		return nil, err
 	}
 
 	if err := parsePQMap(asMap, &uc.PQ); err != nil {
-		return uc, err
+		return nil, err
 	}
 
 	return uc, uc.validate()
 }
 
+func ValidateDefaultVectorDistanceMetric(distance string) error {
```

Comment:
can't this method by private? is it used somewhere else?

Reply:
It is used in usecases/config/config_handler from where I copy pasted this function.

Reply:
function can be renamed to `ValidateVectorDistanceMetric(...)` since distance is now given as an argument. It is not dependant on `config.DefaultVectorDistanceMetric` anymore

Reply:
I renamed it now